### PR TITLE
Correct MessageButtonsBar position (relative to message itself) on screen

### LINKED ISSUE #8463: read confirmation checkmarks obscured by react/reply/more actions ellipsis - cannot get tooltip
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Send a one-line message in talk
2. Try to hover over the check mark(s) to the far right of the message

### Expected behaviour
A tooltip is displayed, like "Message read by everyone who shares their reading status"

Screenshot from 2022-12-12 06-48-21

### Actual behaviour
Can't hover over check mark(s) because a hovering submenu obscures them.

Screenshot from 2022-12-12 06-48-28

## Talk app

**Talk app version:** 15.0.2

**Custom Signaling server configured:** no

**Custom TURN server configured:** no

**Custom STUN server configured:** no

## Browser

**Microphone available:** yes

**Camera available:** yes

**Operating system:** Ubuntu 22.04 LTS 64-bit desktop

**Browser name:** repro's in both latest Firefox and latest Chromium (both installed from snap store)

**Browser version:** Firefox 107.0.1, Chromium 108.0.5359.94

### Browser log

not applicable

## Server configuration

I'm running the `nextcloud:latest` image in a Docker container. Image SHA256 (id) is `60c973f142142baeb88314fb33de083ba1a3967d5774597ecb0bf323fdc0ec52`.

**Operating system**: Docker host is Ubuntu 22.04 LTS 64-bit desktop

**Web server:** Apache

**Database:** SQLite

**PHP version:** 8.1

**Nextcloud Version:** 25.0.2.3

**List of activated apps:**

<details>

```
Enabled:
  - activity: 2.17.0
  - circles: 25.0.0
  - cloud_federation_api: 1.8.0
  - comments: 1.15.0
  - contactsinteraction: 1.6.0
  - dashboard: 7.5.0
  - dav: 1.24.0
  - federatedfilesharing: 1.15.0
  - federation: 1.15.0
  - files: 1.20.1
  - files_pdfviewer: 2.6.0
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - firstrunwizard: 2.14.0
  - logreader: 2.10.0
  - lookup_server_connector: 1.13.0
  - nextcloud_announcements: 1.14.0
  - notifications: 2.13.1
  - oauth2: 1.13.0
  - password_policy: 1.15.0
  - photos: 2.0.1
  - privacy: 1.9.0
  - provisioning_api: 1.15.0
  - recommendations: 1.4.0
  - related_resources: 1.0.3
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - sharebymail: 1.15.0
  - spreed: 15.0.2
  - support: 1.8.0
  - survey_client: 1.13.0
  - systemtags: 1.15.0
  - text: 3.6.0
  - theming: 2.0.1
  - twofactor_backupcodes: 1.14.0
  - updatenotification: 1.15.0
  - user_status: 1.5.0
  - viewer: 1.9.0
  - weather_status: 1.5.0
  - workflowengine: 2.7.0
Disabled:
  - admin_audit
  - bruteforcesettings
  - encryption
  - files_external
  - suspicious_login
  - twofactor_totp
  - user_ldap
```
</details>

**Nextcloud configuration:**

<details>

```
{
    "system": {
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost:8080"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "sqlite3",
        "version": "25.0.2.3",
        "overwrite.cli.url": "http:\/\/localhost:8080",
        "installed": true
    }
}

```
</details>

### Server log (data/nextcloud.log)

none / not applicable

### PR DESCRIPTION
**Pull request by Antreesy**
Signed-off-by: Maksim Sukharev <antreesy.web@gmail.com>

Fix #8463

### 🖼️ Screenshots
:derelict_house:  Before (covers date and check marker, fall outside of visible area): | -
-- | --
image

🏡 After
One- and two-line text messages: | In case buttons bar will be rendered outside of scroll area:
-- | --
image | image

### 🚧 TODO

- [X] Make date visible in right sidebar
- [X] Fix tooltip appearance for date to the left
- [ ] Visual check (both Main Chat and Call Chat in sidebar)
- [ ] Code-review

### 🏁 Checklist

- [X] 📘 API documentation in `docs/` has been updated or is not required
- [X] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [X] 🔖 Capability is added or not needed

### CODE REVIEW

File: src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue [hunk 1]
```diff
@@ -533,12 +533,13 @@ export default {
...
 	display: flex;
 	right: 14px;
-	bottom: -4px;
+	top: calc(50% - $clickable-area / 2);
 	position: absolute;
 	background-color: var(--color-main-background);
 	border-radius: calc($clickable-area / 2);
 	box-shadow: 0 0 4px 0 var(--color-box-shadow);
 	height: 44px;
+	z-index: 1;
```
Comment (nickvergessen) [inline src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:542]: 🙈

**Review by nickvergessen — APPROVED** [review]
Seems to work, not a fan of z-index thou

**Review by marcoambrosini — CHANGES_REQUESTED** [review]
I'm not a fan of the fact that the bar changes position that much depending on the messages heights. Could we put this on hold and think of a better solution in a few days?

File: src/components/MessagesList/MessagesGroup/Message/Message.vue [hunk 2]
```diff
@@ -835,6 +826,20 @@ export default {
...
+				return
+			}
+
+			// Check if the bottom of messageButtonsBar is outside visible area (36px from message bottom)
+			const messageListBottom = this.$parent.$parent.$el.getBoundingClientRect().bottom
+			const buttonsBarSupposedBottom = this.$refs.message.getBoundingClientRect().bottom + 36
+
+			if (messageListBottom < buttonsBarSupposedBottom) {
+				return `bottom: calc(100% - ${this.$refs.messageInfo.offsetTop}px)`
+			}
```
Comment (ShGKme) [inline src/components/MessagesList/MessagesGroup/Message/Message.vue:841]: The solution based on `this.$parent.$parent.$el` in a Vue component depends on other components' implementation in a very implicit way. Maybe we can achieve it in a more explicit solution?

Comment (ShGKme) [reply src/components/MessagesList/MessagesGroup/Message/Message.vue:841]: Also, "36" feels like a magic number here to me. If I understand correctly, it is calculated from the current value of the `$clickable-area` SASS variable. Still, it doesn't depend on the variable value (changing the SASS variable value would require updating this value in the code). If it is equal to the `.message-buttons-bar` bottom property, could we get the element's current value? Or the `--default-clickable-area` CSS variable if it is the same thing.

**Review by marcoambrosini — APPROVED** [review]

**Pull request merged**